docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,382 @@
+
+#### 1.2 Desktop Agent Discovery
+
+Next, attempt to discover whether Desktop Agent Preload or Desktop Agent Proxy interfaces are available, within a specified timeout. The optional `params.timeout` argument to `getAgent()` allows an application to specify the timeout that should be used, with a default value of 750ms. Discovery of Desktop Agent Preload or Desktop Agent Proxy interfaces should be conducted in parallel where possible and the timeout cancelled as soon as an interface is discovered. If a `DesktopAgentDetails` record was found in the previous step, limit discovery to the interface specified, or, if an `agentUrl` property was specified in the `DesktopAgentDetails` record, skip the discovery step entirely and proceed to the next step.
+
+To discover a Desktop Agent Preload interface, check for the presence of an `fdc3` object in the global scope (i.e. `window.fdc3`). If it exists return it immediately. If it does not exist then add a listener for the global `fdc3Ready` event with the the specified timeout, e.g.:
+
+```ts
+const discoverPreloadDA = async (timeout): Promise<DesktopAgent> => {
+  return new Promise((resolve, reject) => {
+    // if the global is already available resolve immediately
+    if (window.fdc3) {
+      resolve(window.fdc3);
+    } else {
+      //`fdc3Ready` event listener function
+      const listener = () => {
+        clearTimeout(timeout);
+        if (window.fdc3) {
+          clearTimeout(timeout);
+          resolve(window.fdc3);
+        } else {
+          reject('The `fdc3Ready` event fired, but `window.fdc3` Was not set!');
+        }
+      };
+      // Setup a timeout to return a rejected promise
+      const timeout = setTimeout(
+        () => {
+          //clear the event listener to ignore a late event
+          window.removeEventListener('fdc3Ready', listener);
+          if (window.fdc3){ 
+            resolve(window.fdc3);
+          } else {
+            reject('Desktop Agent Preload not found ');
+          }
+        }, 
+        timeout
+      );
+      // listen for the fdc3Ready event
+      window.addEventListener('fdc3Ready', listener, { once: true });
+    }
+  });
+};
+```
+
+To discover a Desktop Agent Proxy interface, locate all candidates for a `parent` window or frame by first checking the values of `window.opener` and `window.parent`. If `window.opener !== null` or `window.parent !== window` then they are candidates for a parent window or frame. As iframes can be nested we can also search for candidates that are parents of a parent frame, e.g.:
+
+```ts
+const discoverProxyCandidates = (): WindowProxy[] => {
+  const candidates = [];
+  //parent window
+  if (!!window.opener) { candidates.push(window.opener); }
+
+  //parent frames
+  let currentWin = window;
+  while (currentWin.parent !== currentWin){
```

Comment:
Missing space before opening brace

Reply:
Fixed in 7c089d2bde930a521cf53187abeea8782bac658e